glean-core/android/src/main/java/mozilla/telemetry/glean/private/EventMetricType.kt

```diff
@@ -140,13 +189,29 @@ class EventMetricType<ExtraKeysEnum : Enum<ExtraKeysEnum>>(
      */
     @VisibleForTesting(otherwise = VisibleForTesting.NONE)
     fun testGetValue(pingName: String = sendInPings.first()): List<RecordedEventData> {
-        // @Suppress("EXPERIMENTAL_API_USAGE")
-        // Dispatchers.API.assertInTestingMode()
-
-        // return EventsStorageEngine.getSnapshot(pingName, false)!!.filter { event ->
-        //     event.identifier == identifier
-        // }
-        assert(false, { "Testing API not implementated for EventMetricType" })
-        return listOf(RecordedEventData("LOL", "BAR", 0, null))
+        @Suppress("EXPERIMENTAL_API_USAGE")
+        Dispatchers.API.assertInTestingMode()
+
+        val ptr = LibGleanFFI.INSTANCE.glean_event_test_get_value_as_json_string(
+            Glean.handle,
+            this.handle,
+            pingName
+        )!!
+
+        var jsonRes: JSONArray
+        try {
```

Comment:
Can't this be: `val jsonRes = try {} catch (...) {}`?

Reply:
Good catch.